docsrc/imap/reference/manpages/systemcommands/reconstruct.rst

```diff
@@ -65,70 +65,70 @@ Options
 
 .. program:: reconstruct
 
-.. option:: -C  config-file
+.. option:: -C config-file
 
     |cli-dash-c-text|
 
-.. option:: -p  partition
+.. option:: -p partition, --partition=partition
 
     Search for the listed (non-existant) mailboxes on the indicated
     *partition*. Create the mailboxes in the database in addition to
     reconstructing them. (not compatible with the use of wildcards)
 
-.. option:: -x
+.. option:: -x, --ignore-disk-metadata
 
     When processing a mailbox which is not in the mailbox list (e.g.
     via the **-p** or **-f** options), do not import the metadata from
     the mailbox, instead create it anew (this specifically affects at
     least the mailbox's seen state unique identifier, user flags, and
     ACL).
 
-.. option:: -r
+.. option:: -r, --recursive
 
     Recursively reconstruct all sub-mailboxes of the mailboxes or
     mailbox prefixes given as arguments.
 
-.. option:: -f
+.. option:: -f, --scan-filesystem
 
     Examine the filesystem underneath mailbox, adding all directories
     with a ``cyrus.header`` found there as new mailboxes.  Useful for
     restoring mailboxes from backups.
 
-.. option:: -s
+.. option:: -s, --no-stat
 
     Don't stat underlying files.  This makes reconstruct run faster, at
     the expense of not noticing some issues (like zero byte files or
     size mismatches).  "**reconstruct -s**" should be quite fast.
 
-.. option:: -q
+.. option:: -q, --quiet
 
     Emit less verbose information to syslog.
 
-.. option:: -n
+.. option:: -n, --dry-run
 
     Don't make any changes.  Problems are reported, but not fixed.
 
-.. option:: -G
+.. option:: -G, --force-reparse
 
     Force re-parsing of the underlying message (checks GUID
     correctness). Reconstruct with -G should fix all possible
     individual message issues, including corrupted data files.
 
-.. option:: -I
+.. option:: -I, --update-uniqueids
```

Comment:
Should this rather be named `dedup-uniqueids`?

Reply:
Hmmm... it's hard to say.  I think "dedup-uniqueids" might accidentally imply that it's a system-wide deduplication.  But it's not -- it can only deduplicate them for mailboxes it has visited.  The documentation does make that clear, but the risk with adding long options is that people assume they know what it means from the long option's name, and then don't read the documentation.  So I'm trying to find names that don't accidentally imply the wrong thing.

In that sense, I think the "update-uniqueids" is vague enough that someone who doesn't already know what it means needs to look it up, but memorable enough that someone who does know doesn't need to look it up again.  Maybe...

It's also complicated in that duplicate uniqueids should be impossible under uuid-mailboxes, since how can two mailboxes exist in the same location on disk?  Of course, they can still be duplicated for pairs of older mailboxes that haven't been relocate_by_id'd yet (or for systems that are using mailbox_legacy_dirs).